modules/gapi/src/backends/fluid/gfluidbackend.cpp

```diff
@@ -658,7 +658,22 @@ void cv::gimpl::GFluidExecutable::initBufferRois(std::vector<int>& readStarts,
                     }
 
                     int readStart = resized.y;
-                    cv::gapi::own::Rect roi = adjFilterRoi(resized, fd.border_size, in_meta.size.height);
+                    auto roi = adjFilterRoi(resized, fd.border_size, in_meta.size.height);
+
+                    // Extend required input roi to be even if it's produced by NV12toRGB
+                    if (!in_node->inNodes().empty()) {
+                        auto in_data_producer = in_node->inNodes().front();
+                        if (fg.metadata(in_data_producer).get<FluidUnit>().k.m_kind == GFluidKernel::Kind::NV12toRGB) {
+                            if (roi.y % 2 != 0) {
```

Comment:
What should be even?
- base `.y`
- or `.height`

---

What is about `.width`?

Reply:
Both .y and .height should be even, vertical slicing is not supported for now, so width is not taken into account